airflow_dbt_python/hooks/dbt.py

```diff
@@ -509,43 +505,91 @@ def get_dbt_details_from_connection(self, conn: Connection) -> dict[str, Any]:
 
         return dbt_details
 
+    @classmethod
+    def get_db_hook_class(cls, conn: Connection) -> type[DbtHook]:
+        """Get a dbt hook class depend on Airflow connection type."""
+        known_dbt_hooks = (
+            DbtPostgresHook,
+            DbtRedshiftHook,
+            DbtSnowflakeHook,
+            DbtBigQueryHook,
+            DbtSparkHook,
+        )
+        for hook in known_dbt_hooks:
+            if hook.conn_type == conn.conn_type:
+                return hook
+        return DbtHook
+
 
 class DbtPostgresHook(DbtHook):
     """A hook to interact with dbt using a Postgres connection."""
 
     conn_type = "postgres"
     hook_name = "dbt Postgres Hook"
     conn_params = [
-        DbtConnectionParam("conn_type", "type", "postgres"),
+        DbtConnectionParam("conn_type", "type", conn_type),
         "host",
-        "schema",
+        DbtConnectionParam("schema", default="public"),
         DbtConnectionParam("login", "user"),
         "password",
-        "port",
+        DbtConnectionParam("port", default=5432),
     ]
     conn_extra_params = [
-        "dbname",
-        "threads",
-        "keepalives_idle",
+        DbtConnectionParam("dbname", "database", "postgres"),
         "connect_timeout",
-        "retries",
-        "search_path",
         "role",
+        "search_path",
+        "keepalives_idle",
         "sslmode",
+        "sslcert",
+        "sslkey",
+        "sslrootcert",
+        "retries",
     ]
 
+    @classmethod
+    def get_dbt_details_from_connection(cls, conn: Connection) -> dict[str, Any]:
+        """Extract dbt connection details from Airflow Connection.
+
+        dbt connection details may be present as Airflow Connection attributes or in the
+        Connection's extras. This class' conn_params and conn_extra_params will be used
+        to fetch required attributes from attributes and extras respectively. If
+        conn_extra_params is empty, we merge parameters with all extras.
+
+        Subclasses may override this class attributes to narrow down the connection
+        details for a specific dbt target (like Postgres, or Redshift).
+
+        Args:
+            conn: The Airflow Connection to extract dbt connection details from.
+
+        Returns:
+            A dictionary of dbt connection details.
+        """
+        if "options" in conn.extra_dejson:
+            conn = copy(conn)
+            extra_dejson = conn.extra_dejson
+            options = extra_dejson.pop("options")
+            for k, v in re.findall(r"-c (\w+)=(.*)$", options):
```

Comment:
Could we modify the docstring (or add a comment) explaining what are we trying to find with this regular expression?

Reply:
This is now in a different module, but the same regular expression is present.